bundles/http/Citrix/0b4d1a6b-30f3-4134-a354-42fa008b757e/f3b61af4-efd0-43c2-9a08-a467fc8242c1.js

```diff
@@ -0,0 +1,1029 @@
+                    <Employee>
+                        <PersonIdentity personNumber="${pno.personNumber}"/>
+                    </Employee>
+                </Timesheet>
+            </REQUEST> 
+        </Kronos_WFC>`
+        let {response,result} = await request(body)
+        let path = result.Kronos_WFC.Response[0].Timesheet[0].TotaledSpans[0].TotaledSpan ?? [];
+        if(path.length > 0){
+            let timestampData = path.map(value =>{
+                return {
+                    "date": value.$.Date ? moment(value.$.Date,`${integrationParameters.dateFormat}`).format('yyyy-MM-DD') : null,
+                    "employee_person_identity_person_number": pno.personNumber,
+                    "in_punch_punch_date": value.InPunch[0].Punch[0].$ != undefined && value.InPunch[0].Punch[0].$.Date != undefined ? moment(value.InPunch[0].Punch[0].$.Date,`${integrationParameters.dateFormat}`).format('yyyy-MM-DD') : null,
+                    "in_punch_punch_entered_on_date": value.InPunch[0].Punch[0].$ != undefined && value.InPunch[0].Punch[0].$.EnteredOnDate != undefined ? moment(value.InPunch[0].Punch[0].$.EnteredOnDate,`${integrationParameters.dateFormat}`).format('yyyy-MM-DD') : null,
+                    "in_punch_punch_entered_on_time": value.InPunch[0].Punch[0].$?.EnteredOnTime ?? null ,
+                    "in_punch_punch_entered_kronos_time_zone": value.InPunch[0].Punch[0].$?.KronosTimeZone ?? null,
+                    "in_punch_punch_time": value.InPunch[0].Punch[0].$?.Time ?? null,
+                    "out_punch_punch_date": value.OutPunch[0].Punch[0].$ != undefined && value.OutPunch[0].Punch[0].$.Date != undefined ? moment(value.OutPunch[0].Punch[0].$.Date,`${integrationParameters.dateFormat}`).format('yyyy-MM-DD') : null,
+                    "out_punch_punch_entered_on_date": value.OutPunch[0].Punch[0].$ != undefined && value.OutPunch[0].Punch[0].$.EnteredOnDate != undefined ? moment(value.OutPunch[0].Punch[0].$.EnteredOnDate,`${integrationParameters.dateFormat}`).format('yyyy-MM-DD') : null,
+                    "out_punch_punch_entered_on_time": value.OutPunch[0].Punch[0].$?.EnteredOnTime ?? null,
+                    "out_punch_punch_entered_kronos_time_zone": value.OutPunch[0].Punch[0].$?.KronosTimeZone ?? null,
+                    "out_punch_punch_time": value.OutPunch[0].Punch[0].$?.Time ?? null,
+                    "labor_account_id": value.$.LaborAccountId != undefined ? Number(value.$.LaborAccountId) : null,
+                    "labor_account_name": value.$.LaborAccountName ?? null,
+                    "org_job_id": value.$.OrgJobId != undefined ? Number(value.$.OrgJobId) : null,
+                    "org_job_name": value.$.OrgJobName ?? null,
+                    "unique_key":(++unique_key).toString()
+                }
+            })
+            dataStore.save("record_time_stamp", timestampData);
+        }
+    }
+}
+
+async function syncworkTime(dataStore, client, session, people,integrationParameters){
```

Comment:
We have merged both the functions into one on the basis of `latestSynchronizationTime` value it will determine the functionality. If the `latestSynchronizationTime` is null it behaves as full sync and if `latestSynchronizationTime` is not null it behaves as incremental sync